### PR TITLE
Alerting: fix filtering in notification policies

### PR DESCRIPTION
**Pull request by gillesdemey**
Fixes filtering for notification policies. Previously we were passing `value` instead of `defaultValue` in to the `MatcherFilter` which would make the component render an empty value at all times – effectively disabling user input.

I've removed the `queryString` prop from the component to prevent this from happening in the future.

### CODE REVIEW

File: public/app/features/alerting/unified/components/alert-groups/MatcherFilter.tsx [hunk 1]
```diff
@@ -11,20 +11,22 @@ import { LogMessages } from '../../Analytics';
...
-  queryString?: string;
   defaultQueryString?: string;
   onFilterChange: (filterString: string) => void;
 }
 
-export const MatcherFilter = ({ className, onFilterChange, defaultQueryString, queryString }: Props) => {
+export const MatcherFilter = ({ className, onFilterChange, defaultQueryString }: Props) => {
   const styles = useStyles2(getStyles);
+
   const handleSearchChange = debounce((e: FormEvent<HTMLInputElement>) => {
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/alert-groups/MatcherFilter.tsx:21]: Isn't `debounce` a culprit there? It creates a new `debounced` function on every render

Comment (gillesdemey) [reply public/app/features/alerting/unified/components/alert-groups/MatcherFilter.tsx:21]: That's a fair point, it was a bit wonky but not the culprit – I'll still update this PR to use `useDebounce` instead

**Review by konrad147 — APPROVED** [review]
LGTM!

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM!✨

**Review by VikaCep — APPROVED** [review]

**Comment by gillesdemey** [comment]
No need to back-port to 8.5.x – bug does not occur there.

**Pull request merged**